### PR TITLE
PLANNER-2087: Assign a time budget for the LS to execute

### PR DESCRIPTION
**Pull request by wmedvede**

### CODE REVIEW

**Comment by wmedvede** [comment]
Hi @rsynek , would you mind review this PR?
Thanks.

**Comment by wmedvede** [comment]
Jenkins do full downstream build

File: kie-soup-commons/src/main/java/org/kie/soup/commons/validation/PortablePreconditions.java [hunk 1]
```diff
@@ -145,4 +145,32 @@ public static void checkNullMandatory(final String name,
...
+     * Assert that this parameter is not null and greater than the nonNullValue.
+     * @param name of the parameter.
+     * @param parameter the parameter value.
+     * @param nonNullValue a non null value for executing the comparison.
+     */
+    public static <T> void checkGreaterThan(final String name, final Comparable<T> parameter, final T nonNullValue) {
+        checkNotNull(name, parameter);
+        checkNotNull("nonNullValue", nonNullValue);
+        if (parameter.compareTo(nonNullValue) <= 0) {
+            throw new IllegalArgumentException("Parameter named '" + name + "' must be greater than '" + nonNullValue + "'!");
```
Comment (rsynek) [inline kie-soup-commons/src/main/java/org/kie/soup/commons/validation/PortablePreconditions.java:159]: soft suggestion: let's include also the parameter's value into the exception message.

**Comment by wmedvede** [comment]
FDB failed in non related test, but artifacts were generated

org.kie.hacep.LocalStorageStreamingKieSessionTest.getCommandsTest

Failing for the past 1 build (Since Unstable#1 )
Took 1.7 sec.
Error Message
expected:<2> but was:<1>
Stacktrace
java.lang.AssertionError: expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.kie.hacep.LocalStorageStreamingKieSessionTest.getCommandsTest(LocalStorageStreamingKieSessionTest.java:121)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

**Pull request merged**